src/server.rs

```diff
@@ -321,14 +324,16 @@ async fn classification_with_gen(
 
 async fn generation_with_detection(
     State(state): State<Arc<ServerState>>,
+    headers: HeaderMap,
     WithRejection(Json(request), _): WithRejection<
         Json<models::GenerationWithDetectionHttpRequest>,
         Error,
     >,
 ) -> Result<impl IntoResponse, Error> {
     let request_id = Uuid::new_v4();
     request.validate()?;
-    let task = GenerationWithDetectionTask::new(request_id, request);
+    let filtered_headers = filter_header(&state.orchestrator.get_config(), headers).await;
+    let task = GenerationWithDetectionTask::new(request_id, request, filtered_headers);
```

Comment:
Update per other changes.

```suggestion
    let headers = filter_headers(&state.orchestrator.config().passthrough_headers, headers);
    let task = GenerationWithDetectionTask::new(request_id, request, headers);
```